canine/orchestrator.py

```diff
@@ -415,32 +443,89 @@ def submit_batch_job(self, entrypoint_path, compute_env, extra_sbatch_args = {})
         return batch_id
 
     def job_avoid(self, localizer, overwrite = False): #TODO: add params for type of avoidance (force, only if failed, etc.)
-        # is there preexisting output?
         df_path = localizer.reserve_path(localizer.staging_dir, "results.k9df.pickle")
-        if os.path.exists(df_path.localpath):
-            # load in results and job spec dataframes
-            r_df = pd.read_pickle(df_path.localpath)
-            js_df = pd.DataFrame.from_dict(self.job_spec, orient = "index").rename_axis(index = "_job_id")
-            js_df.columns = pd.MultiIndex.from_product([["inputs"], js_df.columns])
-
-            r_df = r_df.reset_index(col_level = 0, col_fill = "_job_id")
-            js_df = js_df.reset_index(col_level = 0, col_fill = "_job_id")
-
-            # check if jobs are compatible: they must have identical inputs and index,
-            # and output columns must be matching
-            if not r_df["inputs"].columns.equals(js_df["inputs"].columns):
-                raise ValueError("Cannot job avoid; set of input parameters do not match")
-
-            r_df_inputs = r_df[["inputs", "_job_id"]].droplevel(0, axis = 1)
-            js_df_inputs = js_df[["inputs", "_job_id"]].droplevel(0, axis = 1)
 
-            merge_df = r_df_inputs.join(js_df_inputs, how = "outer", lsuffix = "__r", rsuffix = "__js")
-            merge_df.columns = pd.MultiIndex.from_tuples([x.split("__")[::-1] for x in merge_df.columns])
+        # remove all output if specified
+        if overwrite:
+            if os.path.isdir(localizer.staging_dir):
+                rmtree_retry(localizer.staging_dir)
+                os.makedirs(localizer.staging_dir)
+            return 0
 
-            if not merge_df["r"].equals(merge_df["js"]):
-                raise ValueError("Cannot job avoid; values of input parameters do not match")
-
-            # if jobs are indeed compatible, we can figure out which need to be re-run
-            r_df.loc[r_df[("job", "slurm_state")] == "FAILED"]
-
-            # destroy their output directories
+        # check for preexisting jobs' output
+        if os.path.exists(df_path.localpath):
```

Comment:
Why are we only loading the local path? This should be using the transport API to check if the file exists on the controller.

```
if transport.exists(df_path.controllerpath)
```

Reply:
Cuz I'm a short-sighted idiot who only wrote this with the NFS localizer in mind? ;-)